server/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/SamplerAggregator.java

```diff
@@ -56,6 +58,8 @@
     public static final ParseField MAX_DOCS_PER_VALUE_FIELD = new ParseField("max_docs_per_value");
     public static final ParseField EXECUTION_HINT_FIELD = new ParseField("execution_hint");
 
+    static final long SCOREDOCKEY_SIZE = RamUsageEstimator.shallowSizeOfInstance(DiversifiedTopDocsCollector.ScoreDocKey.class);
```

Comment:
@markharwood I went ahead and evaluated the full object using `shallowSizeOfInstance()` instead of stringing together constants.  I know it's a bit slower (reflection etc) but thought it might be ok since the value is a static final?

I don't have much experience with `RamUsageEstimator` though, so happy to switch back to adding together constants if that's better.

Reply:
I don't have experience with `shallowSizeOfInstance` either but I see it used in several places in core and use as static final makes sense. I know it has some stuff in there to account for compressed vs uncompressed references.